src/components/LessonCodeSnippet.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { capitalize } from '../helpers'
+
+const Snippet = (props) => {
+  return (
+    <div className={'lesson-module-code__code' + (props.active ? ' lesson-module-code__code--active' : '')} id={props.id}>
+      <pre><code className={props.type + ' hljs'}>{props.code}</code></pre>
+    </div>
+  )
+}
+
+const CodeHeader = ({ id, title, active, onClick }) => {
+  return (
+    <a key={id}
+      onClick={() => onClick(id)}
+      className={'lesson-module-code__trigger ' + (active ? ' lesson-module-code__trigger--active' : '')}
+      data-target={id}>{title}
+    </a>
+  )
+}
+
+class LessonCodeSnippet extends React.Component {
+  constructor(props) {
+    super(props)
+    this.onClick = this.onClick.bind(this)
+    this.getType = this.getType.bind(this)
+    this.getTItle = this.getTitle.bind(this)
+    this.state = {
+      active: 'javascript'
+    }
+  }
+
+  onClick(id) {
+    this.setState({
+      active: id
+    })
+  }
+
+  getType(platform) {
+    const types = {
+      curl: 'shell',
+      dotNet: 'csharp',
+      android: 'javaAndroid'
+    }
+    return (types[platform]) ? types[platform] : platform
+  }
+
+  getTitle(platform) {
+    const titles = {
+      curl: 'cURL',
+      dotNet: '.NET',
+      javascript: 'JavaScript',
+      php: 'PHP'
+    }
+    return (titles[platform]) ? titles[platform] : capitalize(platform)
+  }
+
+  render() {
+    const headers = []
+    const snippets = []
+    for (let prop in this.props) {
```

Comment:
It's not a big deal for now but this may cause problems in the future if we introduce another property different than `__typename` and `title` and not intended for the code snippet.
Maybe add some `Proptypes` to make sure that the next developer working with this will get warned when there is more data

Reply:
I totally agree, would be better if the platforms were wrapped in a parent object or something. for now I'll add proptypes and a comment